apps/server/src/api-data/settings/settings.controller.ts

```diff
@@ -65,3 +66,14 @@ export async function postSettings(req: Request, res: Response<Settings | ErrorR
     res.status(400).send({ message });
   }
 }
+
+export async function getWelcomeDialog(req: Request, res: Response) {
+  const show = await appState.getShowWelcomeDialog();
+  res.status(200).send({ show });
+}
+
+export async function postWelcomeDialog(req: Request, res: Response) {
+  const { show } = req.body as unknown as { show: boolean };
+  await appState.setShowWelcomeDialog(show);
+  res.status(200).send();
```

Comment:
suggestion: not sure if the downcasting is doing much for here, we could simplify like

```ts
  await appState.setShowWelcomeDialog(req.body.show as boolean);
```